federation/pkg/dnsprovider/providers/aws/route53/rrsets.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package route53
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+)
+
+type ResourceRecordSets struct {
+	zone *Zone
+}
+
+func (rrsets ResourceRecordSets) List() ([]dnsprovider.ResourceRecordSet, error) {
+	input := route53.ListResourceRecordSetsInput{
+		HostedZoneId: rrsets.zone.impl.Id,
+	}
+	response, err := rrsets.zone.zones.interface_.service.ListResourceRecordSets(&input)
+	// TODO: Handle truncated responses
+	if err != nil {
+		return nil, err
+	}
+	list := make([]dnsprovider.ResourceRecordSet, len(response.ResourceRecordSets))
+	for i, rrset := range response.ResourceRecordSets {
+		list[i] = &ResourceRecordSet{rrset, &rrsets}
+	}
+	return list, nil
+}
+
+func (rrsets ResourceRecordSets) Add(rrset dnsprovider.ResourceRecordSet) (dnsprovider.ResourceRecordSet, error) {
```

Comment:
Agreed in principle.  My only defence is that ResourceRecordSets is a stateless interface of methods.  But I guess it is mildly confusing.  I'll address in a separate PR, as this again has knock-on effects on other libraries.